src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_jni.h

```diff
@@ -465,6 +465,40 @@ extern jmethodID g_KeyAgreementInit;
 extern jmethodID g_KeyAgreementDoPhase;
 extern jmethodID g_KeyAgreementGenerateSecret;
 
+// Compatibility macros
+#if !defined (__mallocfunc)
+#if defined (__clang__) || defined (__GNUC__)
+#define __mallocfunc __attribute__((__malloc__))
+#else // def (__clang__ || __GNUC__)
+#define __mallocfunc
+#endif // ndef (__clang__ || __GNUC__)
+#endif
+
+#if !defined (__BIONIC_ALLOC_SIZE)
+#if defined (__clang__) || defined (__GNUC__)
+#define __BIONIC_ALLOC_SIZE(...) __attribute__((__alloc_size__(__VA_ARGS__)))
+#else // def (__clang__ || __GNUC__)
+#define __BIONIC_ALLOC_SIZE(...)
+#endif // ndef (__clang__ || __GNUC__)
+#endif
+
+#if !defined (__wur)
+#if defined (__clang__) || defined (__GNUC__)
+#define __wur __attribute__((__warn_unused_result__))
+#else // def (__clang__ || __GNUC__)
+#define __wur
+#endif // ndef (__clang__ || __GNUC__)
+#endif
+
+#if defined (__clang__) || defined (__GNUC__)
+#define ARGS_NON_NULL(...) __attribute__((nonnull (__VA_ARGS__)))
+#else
+#define ARGS_NON_NULL_ALL
+#define ARGS_NON_NULL(_idx1_, ...)
```

Comment:
missed it earlier, but would probably be good to include a comment that the index is 1-based, not 0-based.

Reply:
I'll add the comment in the next PR, if you don't mind - don't want to lose all that awesome greenness below :)